### ISSUE TITLE
How to version modules in IETF drafts (after they have been published at 1.0.0 or later)

### ISSUE THREAD

**Issue by rgwilton** [netmod-wg/yang-ver-dt#10]
Somewhat related to #9, how would a YANG module be versioned in an IETF draft after is has previously been published, but before a new version is published.

E.g. say RFC 8343 was published with semver 1.0.0, and then we want to do 8343-bis.  What versions do we put in the 8343-bis drafts?  Do these need to follow the strict semver rules, that could be kind of painful?  Or perhaps the recommendation could be just that they use 1.0.1m, 1.0.2m, etc for each revision, and then when finally published they are assigned version 1.1.0 or 2.0.0 depending on whether the changes are BC or NBC.

**Comment by BalazsLengyel** [comment]
For drafts in Ericsson we go from
      1.0.0 to the next needed revision 1.0.1 or 1.1.0 or 2.0.0
      immediately. 
      Later during the review process, when many updates may be made, 
      we only increase the patch level, so e.g. we might end up at
      2.0.27 
      if we have 27 unreleased updates.
    regards Balazs
    
    On 2019. 03. 07. 18:15, Robert Wilton
      wrote:
    
    
      
      Somewhat related to #9, how would a YANG module be
        versioned in an IETF draft after is has previously been
        published, but before a new version is published.
      E.g. say RFC 8343 was published with semver 1.0.0, and then we
        want to do 8343-bis. What versions do we put in the 8343-bis
        drafts? Do these need to follow the strict semver rules, that
        could be kind of painful? Or perhaps the recommendation could be
        just that they use 1.0.1m, 1.0.2m, etc for each revision, and
        then when finally published they are assigned version 1.1.0 or
        2.0.0 depending on whether the changes are BC or NBC.
      —
        You are receiving this because you are subscribed to this
        thread.
        Reply to this email directly, view it on GitHub, or mute the thread.
      {"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/netmod-wg/yang-ver-dt","title":"netmod-wg/yang-ver-dt","subtitle":"GitHub repository","main_image_url":"https://github.githubassets.com/images/email/message_cards/header.png","avatar_image_url":"https://github.githubassets.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/netmod-wg/yang-ver-dt"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"How to version modules in IETF drafts (after they have been published at 1.0.0 or later) (#10)"}],"action":{"name":"View Issue","url":"https://github.com/netmod-wg/yang-ver-dt/issues/10"}}}
      [
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/netmod-wg/yang-ver-dt/issues/10",
"url": "https://github.com/netmod-wg/yang-ver-dt/issues/10",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]
    
    -- 
Balazs Lengyel                       Ericsson Hungary Ltd.
Senior Specialist
Mobile: +36-70-330-7909              email: Balazs.Lengyel@ericsson.com

**Comment by xorrkaz** [comment]
Text has been added to the yang-semver to address module development within the IETF:

```
   When an existing IETF module is being revised, it MUST use the target
   version for the revision-label with a pre-release string that
   includes the current RFC number plus the string "bis".  For example,
   if the module defined in RFCXXXX at version 1.0.0 is being revised to
   include non-backwards-compatible changes, its development revision-
   labels MUST include 2.0.0-XXXXbis.  Since they MUST also be unique,
   additional alphanumeric identifiers MUST be used (e.g., 2.0.0-
   XXXXbis-dev1).  Since each new bis will work off a new RFC number,
   this nomenclature ensures uniqueness for the module throughout its
   lifecycle.

   If a module is being revised and the original module never had a
   revision-label (i.e., you wish to start using YANG semver in future
   module revisions), choose a semver value that makes the most sense
   based on the module's history.  For example, if a module started out
   in the pre-NMDA world and then had NMDA support added without
   removing any legacy "state" branches, and you are looking to add
   additional new features, a sensible choice for the target YANG semver
   would be 1.2.0 (since 1.0.0 would have been the initial, pre-NMDA
   release, and 1.1.0 would have been the NMDA revision).
```
